src/number_theory/pell.lean

```diff
@@ -137,4 +137,119 @@ end
 
 end existence
 
+/-!
+### Group structure of the solution set
+
+We define a structure of a commutative multiplicative group with distributive negation
+on the set of all solutions to the Pell equation `x^2 - d*^2 = 1`.
+
+The type of such solutions is `pell.solution₁ d`. It contains integers `x` and `y` and
+a proof that `(x, y)` is indeed a solution.
+
+The multiplication is given by `(x, y) * (x', y') = (x*y' + d*y*y', x*y' + y*x')`
+(this is obtained by mapping `(x, y)` to `x + y*√d` and multiplying the results).
+-/
+
+-- We use `solution₁ d` to allow for a more general structure `solution d m` that
+-- encodes solutions to `x^2 - d*y^2 = m` to be added later.
```

Comment:
Why not just go straight to the general version, but for now only include the instances for `solution d 1`?

Reply:
Trying to work with `solution d 1` gives me `motive not type correct` trouble when I try to use `a.rel`.
Since the case with right hand side 1 is of special interest, I think it should be OK to leave it as is right now (and then later prove a suitable isomorphism between `solution d 1` and `solution₁ d`).

Reply:
I'm a bit surprised by this error message, but I guess it's moot with the change to `unitary` anyway.